acme_diags/driver/streamflow_driver.py

```diff
@@ -0,0 +1,370 @@
+            # Estimated drainage area (km^2) from observed
+            area_observed = gauges[i, 13].astype(numpy.float64)
+
+            if area_upstream is not None:
+                drainage_area_error, lat_lon_observed = get_drainage_area_error(
+                    radius, resolution, lon_observed, lat_observed, area_upstream, area_observed)
+            else:
+                # Use the center location
+                lat_lon_observed = [lat_observed, lon_observed]
+            if parameter.ref_mat_file:
+                origin_id = gauges[i, 1].astype(numpy.int64)
+                # Column 0 -- year
+                # Column 1 -- month
+                # Column origin_id + 1 -- the observed streamflow from gauge with the corresponding origin_id
+                extracted = observed[:, [0, 1, origin_id + 1]]
+                monthly_mean = numpy.zeros((12,1)) + numpy.nan
+                for month in range(12):
+                    # Add 1 to month to account for the months being 1-indexed
+                    if sum(extracted[:, 1] == month + 1) > 0:
+                        # `extracted[:,1]`: for all x, examine `extracted[x,1]`
+                        # `extracted[:,1] == m`: Boolean array where 0 means the item in position [x,1] is NOT m,
+                        # and 1 means it is m
+                        # Example:
+                        # a = [[1,2,3,4],
+                        #      [5,6,7,8],
+                        #      [1,2,3,4]]
+                        # a[:,1]: [[2],
+                        #          [6],
+                        #          [2]]
+                        # a[:,1] == 2: [[1], # False is 0, True is 1
+                        #               [0],
+                        #               [1]]
+                        # a[a[:,1] == 2, 2]: [[3],
+                        #                     [3]]
+                        monthly_mean[month] = numpy.nanmean(extracted[extracted[:, 1] == month + 1, 2])
+                # This is observed annual mean streamflow
+                annual_mean = numpy.mean(monthly_mean)
+            if parameter.ref_mat_file and numpy.isnan(annual_mean):
+                # All elements of row i will be nan
+                export[i,:] = numpy.nan
+            else:
+                if parameter.ref_mat_file:
+                    # Reshape extracted[:,2] into a 12 x ? matrix; -1 means to calculate the size of the missing dimension.
+                    mmat = numpy.reshape(extracted[:, 2], (12, -1))
+                    mmat_id = numpy.sum(mmat, axis=0).transpose()
+                    if numpy.sum(~numpy.isnan(mmat_id), axis=0) > 0:
+                        # There's at least one year of record
+                        monthly = mmat[:, ~numpy.isnan(mmat_id)]
+                    else:
+                        monthly = monthly_mean
+                    seasonality_index_observed, peak_month_observed = get_seasonality(monthly)
+                else:
```

Comment:
If we're not using a `mat` file for the `obs`, then just treat the `obs` exactly the same as the `sim`. Is that the right idea?

Reply:
Yes, for run_type = `model_vs_model`, both datasets should be treated the same.